### PR TITLE
Fix conversion of list and maps in Yaml

### LINKED ISSUE #60: yaml to java conversion fails for ArrayList
### Issue description

Clone branch `security-tranining` at `micronaut-advocacy/micronaut-live-newsletter`

https://github.com/micronaut-advocacy/micronaut-live-newsletter/tree/security-training

run 

`./gradlew :app:prepareJitOptimizations`

```13:40:12.672 [main] INFO  i.m.a.s.s.YamlPropertySourceGenerator - Converting application.yml into Java based configuration
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:118)
	at io.micronaut.aot.cli.Main.run(Main.java:92)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at io.micronaut.aot.cli.Main.execute(Main.java:139)
	at io.micronaut.aot.cli.Main.main(Main.java:143)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:112)
	... 10 more
Caused by: java.lang.UnsupportedOperationException: Configuration map contains an entry of type class java.util.ArrayList which is not supported yet. Please file a bug report.
Caused by: java.lang.reflect.InvocationTargetException

	at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generateMap(MapPropertySourceGenerator.java:95)
	at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generate(MapPropertySourceGenerator.java:116)
	at io.micronaut.aot.core.codegen.AbstractSingleClassFileGenerator.generate(AbstractSingleClassFileGenerator.java:35)
	at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.createMapProperty(YamlPropertySourceGenerator.java:75)
	at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.generate(YamlPropertySourceGenerator.java:58)
Caused by: java.lang.UnsupportedOperationException: Configuration map contains an entry of type class java.util.ArrayList which is not supported yet. Please file a bug report.

	at io.micronaut.aot.std.sourcegen.AbstractStaticServiceLoaderSourceGenerator.generate(AbstractStaticServiceLoaderSourceGenerator.java:104)
	at io.micronaut.aot.core.codegen.ApplicationContextConfigurerGenerator.generate(ApplicationContextConfigurerGenerator.java:59)
	at io.micronaut.aot.MicronautAotOptimizer$Runner.execute(MicronautAotOptimizer.java:318)
	at io.micronaut.aot.MicronautAotOptimizer.execute(MicronautAotOptimizer.java:191)
	... 15 more

```

### PR DESCRIPTION
**Pull request by melix**
The map converter now supports lists and maps values.

Fixes #60

### CODE REVIEW

File: aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy [hunk 1]
```diff
@@ -47,9 +51,45 @@ public class Test_configStaticPropertySource extends MapPropertySource {
         put("micronaut.application.name", "demoApp");
         put("micronaut.server.port", 8181);
         put("micronaut.server.cors.enabled", true);
+        put("micronaut.security.intercept-url-map", list0());
         }});
   }
 
+  private static List list2() {
+    List result = new ArrayList<>();
```
Comment (graemerocher) [inline aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy:59]: can these arrays be sized to the size of the data?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy (hunk 1 above)
Comment (melix) [reply aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy]: done

File: aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy [hunk 2]
```diff
@@ -56,35 +57,31 @@ public class Test_configStaticPropertySource extends MapPropertySource {
...
+    Map result = new LinkedHashMap<>(3);
     result.put("pattern", "/api/v1/subscriber/count");
     result.put("http-method", "GET");
     result.put("access", list4());
     return result;
   }
 
   private static List list0() {
-    List result = new ArrayList<>();
+    List result = new ArrayList<>(2);
```
Comment (graemerocher) [inline aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy:84]: One other nice optimizations would be to use `List.of` on JDK 11+ :)

Comment (melix) [reply aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy:84]: Yes, I wish I could do this now... but for now I decided to keep it simple stupid :) With Micronaut 4 we can do this.

**Pull request merged**